api/net/Areas/Admin/Controllers/FolderController.cs

```diff
@@ -0,0 +1,130 @@
+using System.Net;
+using System.Net.Mime;
+using System.Text.Json;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Swashbuckle.AspNetCore.Annotations;
+using TNO.API.Areas.Admin.Models.Folder;
+using TNO.API.Models;
+using TNO.DAL.Services;
+using TNO.Keycloak;
+
+namespace TNO.API.Areas.Admin.Controllers;
+
+/// <summary>
+/// FolderController class, provides folder endpoints for the api.
+/// </summary>
+[ClientRoleAuthorize(ClientRole.Administrator)]
+[ApiController]
+[Area("admin")]
+[ApiVersion("1.0")]
+[Route("api/v{version:apiVersion}/[area]/folders")]
+[Route("api/[area]/folders")]
+[Route("v{version:apiVersion}/[area]/folders")]
+[Route("[area]/folders")]
+[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Unauthorized)]
+[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Forbidden)]
+public class FolderController : ControllerBase
```

Comment:
@jdtoombs This is almost exactly what you'll need when you create the endpoints for for subscriber folders.